plugin/src/main/java/com/denizenscript/denizen/objects/NPCTag.java

```diff
@@ -1277,6 +1277,98 @@ else if (attribute.startsWith("list", 2)) {
             return result;
         });
 
+        // <--[tag]
+        // @attribute <NPCTag.waypoint_provider>
+        // @returns ElementTag
+        // @description
+        // Returns the NPC's current Waypoint Provider type.
+        // -->
+        tagProcessor.registerTag(ElementTag.class, "waypoint_provider", (attribute, object) -> {
+            if (object.getCitizen().hasTrait(Waypoints.class)) {
```

Comment:
use getTraitNullable + a null check (if you found other code written the way this is, that's due to Citizens previous not having the nullable variant so old code isn't all updated for it)